### ISSUE TITLE
SC problematic for Closed functionality: SC 2.4.7 Focus Visible

### ISSUE THREAD

**Issue by pday1** [w3c/wcag2ict#271]
From closed functionality questionnaire: 
https://www.w3.org/2002/09/wbs/55145/wcag2ict-sc-problematic-for-closed/results#xq26

Phil Day | 2.4.7 Focus Visible. Some closed systems use tactilely discernible input such as a numeric keyboard to enter information in conjunction with private audio prompts. In this case, there is no change of 'focus' as there is no navigation around visible elements. The example I am thinking of is an ATM or kiosk where somebody using the private audio hears "For cash withdrawal, press 1; for deposits, press 2..." and then presses numbers in response. I would argue that this shouldn't meet the focus requirement, but think we might need to add a note to this effect.

Mike Pluke | Phil raises a valid point. This means of locating controls on a numeric keypad may also be applicable to keys that "form part of a functional group or are strategically positioned so that they can be tactiley discerned" to paraphrase a suggested similar proposed change to a closed functionality equivalent requirement in EN 301 549. For example, the "Clear" and "Cancel" keys on an ATM are usually positioned to allow easy non-visual location and might be referred to in a spoken presentation of on-screen instructions.

SC text included for reference
> SC 2.4.7 Focus Visible
> Any keyboard operable user interface has a mode of operation where the keyboard focus indicator is visible.
>

**Comment by pday1** [comment]
Proposed content for SC problematic for Closed functionality section:

[2.4.7 Focus Visible ](#focus-visible) - Requires operation where focus can be moved and controlled by keyboard. Some closed systems may offer tactilely discernible input such as a numeric keypad or other functional groups of keys, but do not offer any change of focus; for example, the keys are used to select options from a spoken menu rather than to move an onscreen focus element between multiple options. In this case, there is no concept of focus, and therefore there is no need for a visible indicator.

**Comment by pday1** [comment]
For ease of review, here is content pulled from the latest Editor's draft to show the original "Applying SC to Non-Web..." alongside the proposed new content for "SC problematic for Closed...".

The latest editor's draft can be found at https://w3c.github.io/wcag2ict/
(Note this does not contain content that is in pull requests that are open and not yet approved)

**Content from Applying SC**

https://w3c.github.io/wcag2ict/#applying-sc-2-4-7-focus-visible-to-non-web-documents-and-software

> This applies directly as written, and as described in [Intent from Understanding Success Criterion 2.4.7](https://www.w3.org/WAI/WCAG22/Understanding/focus-visible#intent) (also provided below).


**Latest proposed content for SC problematic for Closed functionality section**

> [2.4.7 Focus Visible](https://w3c.github.io/wcag2ict/#focus-visible) - Requires a mode of operation where focus can be moved and controlled by keyboard. Some closed systems may offer tactilely discernible input such as a numeric keypad or other functional groups of keys, but do not offer any mechanism for focus; for example, the keys are used to select options from a spoken menu rather than to move an onscreen focus element between multiple options. In this case, there is no concept of focus, and therefore there is no need for a visible indicator.

**Comment by maryjom** [comment]
From the [Results from last survey](https://www.w3.org/2002/09/wbs/55145/WCAG2ICT-SC-problematic-last-5/results#xq8) that closed 13 Dec. 2023, three additional proposals emerged:
**Option 1:  Original proposal + edits**
>2.4.7 Focus Visible - Requires a mode of operation enabling focus movement and control by keyboard. While certain closed systems may provide tactile input like numeric keypads or other functional groups of keys, they lack a mechanism for traditional onscreen focus. For example, keys might be used to select options from a spoken menu rather than navigating an onscreen focus element between multiple options. In such cases, the absence of a conventional focus eliminates the need for a visible indicator.

**Option 2: to eliminate "requires mode of operation enabling ...control by keyboard**
>This Success Criterion assumes there is a user interface component that receives keyboard focus. Menu-driven interfaces do not necessarily have visible components that receive focus, as they may have tactilely discernible input keys used to choose from a spoken menu of options. In such cases, there is no concept of focus or need for a visible focus indicator.

**Option 3: Add general guidance Note for 2.4.7 + bullet for SC problematic for closed**
>General guidance note:
>NOTE: This criterion applies where there are user interface controls that are focusable using a keyboard interface. Some software may accept input from a keyboard, keypad, or controller, yet not display user interface controls; for example, the keys are mapped directly to functions without activating on-screen controls. In this case, there is no concept of focus and this criterion is met.
>Bullet for SC problematic for closed:
>* 2.4.7 Focus Visible - As noted in the section [Applying SC 2.4.7 Focus Visible to Non-Web Documents and Software](), when there are no on-screen controls this criterion is met. In addition, when a product with closed functionality has on-screen controls (for example, buttons on a touchscreen) and does not have a keyboard interface, there is no concept of keyboard focus and this criterion is met; however, as noted for 2.1.1 Keyboard it would need an alternate way to access all functionality.

**Comment by maryjom** [comment]
In the 7 March meeting the TF made a [resolution on 2.4.7](https://www.w3.org/2024/03/07-wcag2ict-minutes#r03) to use an edited version of the surveyed Option 4 that reads:
- 2.4.7 Focus Visible - Presumes that there is a mode of operation where focus can be moved and controlled by keyboard. Some closed systems may offer tactilely discernible input such as a numeric keypad or other functional groups of keys, but do not offer any mechanism for conveying focus because the user interface is designed not to need that. For example, the keys are used to select options from a spoken menu rather than to move an onscreen focus element between multiple options. In this case, there is no concept of focus, thus there is no need for a visible indicator and this Success Criterion would be satisfied.

**Comment by maryjom** [comment]
The agreed content in PR #318 has been merged into the editor's draft, so this can be closed.
